### DISCUSSION TITLE
<CR> binding for autocompletion in Lua

### DISCUSSION THREAD

**Discussion by tomkoid** [neoclide/coc.nvim#4048] [Q&A] [answered]
Recently I rewrote my NeoVim config to Lua and after recent update the autocompletion didn't work. I figured out, that I need to manually bind the <CR> key with some Vim Script code. It worked, but for me it doesn't make sense to have the entire config written in Lua except for this and I don't understand Vim Script that much. How can I write this in Lua?

```
function! s:check_back_space() abort
  let col = col(".") - 1
  return !col || getline(".")[col - 1] =~ "\s"
endfunction

if has('nvim')
  inoremap <silent><expr> <c-space> coc#refresh()
else
  inoremap <silent><expr> <c-@> coc#refresh()
endif

inoremap <expr> <cr> coc#pum#visible() ? coc#_select_confirm() : "\<CR>"
```

**Comment by chemzqm** [comment, marked as answer]
Checkout  nvim's doc `:h vim.keymap.set()`
